core/src/main/kotlin/org/evomaster/core/output/naming/ActionTestCaseNamingStrategy.kt

```diff
@@ -51,6 +62,34 @@ abstract class ActionTestCaseNamingStrategy(
         } else {
             addActionResult(individual.evaluatedMainActions().last(), nameTokens)
         }
+        addEnvironmentActions(individual, nameTokens)
+    }
+
+    private fun addEnvironmentActions(individual: EvaluatedIndividual<*>, nameTokens: MutableList<String>) {
+        val initializingActions = individual.individual.seeInitializingActions()
+        val allActions = individual.individual.seeAllActions()
+
+        val initActionNames = mutableListOf<String>()
+        if (hasSqlAction(initializingActions)) initActionNames.add(sql)
+        if (hasMongoAction(initializingActions)) initActionNames.add(mongo)
+        if (usesWireMock(allActions)) initActionNames.add(wiremock)
+
+        if (initActionNames.isNotEmpty()) {
+            nameTokens.add(using)
+            nameTokens.addAll(initActionNames)
+        }
+    }
+
+    private fun hasSqlAction(actions: List<EnvironmentAction>): Boolean {
+        return actions.any { it is SqlAction }
+    }
+
+    private fun hasMongoAction(actions: List<EnvironmentAction>): Boolean {
+        return actions.any { it is MongoDbAction }
+    }
+
+    private fun usesWireMock(actions: List<Action>): Boolean {
+        return config.isEnabledExternalServiceMocking() && actions.any { it is HttpExternalServiceAction }
```

Comment:
hi. just noticed it... but, why the `config.isEnabledExternalServiceMocking()`?  if there is any action of type `HttpExternalServiceAction`, then it is implied that WM support is enabled.  isn't  `config.isEnabledExternalServiceMocking()` redundant?

Reply:
@arcuri82, took that from the TestSuiteWriter file here: https://github.com/WebFuzzing/EvoMaster/blob/master/core/src/main/kotlin/org/evomaster/core/output/service/TestSuiteWriter.kt#L397

Which is, now that I remember, from where I also took the usage of the `HostnameResolutionAction`, since the `needWireMockServers` checks for that: https://github.com/WebFuzzing/EvoMaster/blob/master/core/src/main/kotlin/org/evomaster/core/search/Solution.kt#L46

So, if it's redundant, I'll remove it, not very sure from my end

Reply:
yep, I think it is redundant.  `config.isEnabledExternalServiceMocking()` would imply that WM can be used... but that does not mean that necessarely a test is using it. from a logical perspective, we have a case of `A && B`, where `notA => notB` and `B => A`. so `A` is redundant when checking `A && B` (isn't it?)

Reply:
Fair enough, yes. From a logical perspective I fully agree. My next two questions are:
1. Why does `TestSuiteWriter` check both conditions?
2. Why does `Solution.needWireMockServers` use `HostnameResolutionAction` instead of `HttpExternalServiceAction`?

Reply:
regarding [TestSuiteWriter.kt#L397](https://github.com/WebFuzzing/EvoMaster/blob/master/core/src/main/kotlin/org/evomaster/core/output/service/TestSuiteWriter.kt#L397), those are import statements at the beginning of the test suite. you just need 1 test in whole suite to require executing that code. Indeed `config.isEnabledExternalServiceMocking()` could be considered redundant here, but, it is a cheaper check then looking at whole `solution` (so, if false, it is quicker to compute)

Regarding `solution.needWireMockServers()` might indeed rather check `HttpExternalServiceAction`. that might be old check from time before we had a sink address.  but worst case it just adds imports that are not used. 
@seran  can you comment on this?

thx

Reply:
Got it, I'll remove the `config.isEnabledExternalServiceMocking()` and wait for the comments regarding `HttpExternalServiceAction`

Reply:
1. `isEnabledExternalServiceMocking()` is used in several other places as a setting to enable or disable functions related to external service handling.
2. There could be cases where we only need to reroute the external service call to the network sinkhole. In such instances, we use `HostnameResolutionAction` to configure the DNS information, and there will be no `HttpExternalServiceAction `.
3. If the test acts upon the response from the external service, then we use the `HttpExternalServiceAction` to configure the stubs, along with `HostnameResolutionAction` to configure DNS information.

As the name suggests, if your method only needs to check for WM usage, then the current logic seems fine without the `isEnabledExternalServiceMocking()` check.

You can assume that if there is a `HttpExternalServiceAction`, there is a `HostnameResolutionAction ` but not vice versa.
Also, all of this is will happen if the `isEnabledExternalServiceMocking()` returns `true.`

@arcuri82 we need to do a clean-up in other places where we have the same check as `isEnabledExternalServiceMocking()`.

Reply:
Understood, thanks @seran for the explanation! I believe then we're good to go with the review @arcuri82, I'll leave the `HttpExternalServiceAction` as the check for WM